homeassistant/components/linkplay/media_player.py

```diff
@@ -0,0 +1,240 @@
+) -> None:
+    """Set up a media player from a config entry."""
+
+    @callback
+    def add_entity(bridge: LinkPlayBridge) -> None:
+        async_add_entities([LinkPlayMediaPlayerEntity(bridge)])
+
+    entry.async_on_unload(async_dispatcher_connect(hass, BRIDGE_DISCOVERED, add_entity))
+
+
+class LinkPlayMediaPlayerEntity(MediaPlayerEntity):
+    """Representation of a LinkPlay media player."""
+
+    _bridge: LinkPlayBridge
+
+    def __init__(self, bridge: LinkPlayBridge) -> None:
+        """Initialize the LinkPlay media player."""
+        self._bridge = bridge
+
+        self._attr_unique_id = self.entity_id = f"{DOMAIN}.{self._bridge.device.uuid}"
+        self._attr_name = self._bridge.device.name
+        self._attr_sound_mode_list = list(EQUALIZER_MAP.values())
+        self._attr_should_poll = True
+        self._attr_device_class = MediaPlayerDeviceClass.RECEIVER
+        self._attr_source_list = [
+            SOURCE_MAP[playing_mode]
+            for playing_mode in self._bridge.device.playmode_support
+        ]
+        self._attr_media_content_type = MediaType.MUSIC
+
+    @property
+    def device_info(self) -> dr.DeviceInfo:
+        """Return the device info."""
+        manufacturer, model = get_info_from_project(
+            self._bridge.device.properties["project"]
+        )
+        return dr.DeviceInfo(
+            configuration_url=self._bridge.endpoint,
+            connections={
+                (dr.CONNECTION_NETWORK_MAC, self._bridge.device.properties["MAC"])
+            },
+            entry_type=None,
+            hw_version=self._bridge.device.properties["hardware"],
+            identifiers={(DOMAIN, self._bridge.device.uuid)},
+            manufacturer=manufacturer,
+            model=model,
+            name=self._bridge.device.name,
+            suggested_area=None,
+            sw_version=self._bridge.device.properties["firmware"],
+            via_device=(DOMAIN, DOMAIN),
+        )
+
+    @property
+    def group_members(self) -> list[str]:
+        """List of members which are currently grouped together."""
+
+        multiroom = get_active_multiroom(self.hass, self._bridge)
+        if multiroom is None:
+            return []
+
+        uuids = [multiroom.leader.device.uuid] + [
+            follower.device.uuid for follower in multiroom.followers
+        ]
+
+        uuids.remove(self._bridge.device.uuid)
+
+        return uuids
+
+    async def async_update(self) -> None:
+        """Update the state of the media player."""
+        try:
+            await self._bridge.player.update_status()
+            self._update_properties()
+        except LinkPlayException as e:
+            _LOGGER.error(e)
+            self._attr_available = False
+
+    async def async_select_source(self, source: str) -> None:
+        """Select input source."""
+        await self._bridge.player.set_play_mode(SOURCE_MAP_INV[source])
```

Comment:
Can this raise a `LinkPlayException` ? If so, let's create a decorator around the command methods to turn those into HomeAssistantError

Reply:
All API calls raise `LinkPlayRequestException` on error so I'll re-raise them as `HomeAssistantError` 👍